src/planner/parabolic/HauserParabolicSmootherHelpers.cpp

```diff
@@ -0,0 +1,198 @@
+#include "ParabolicUtil.hpp"
+#include "Config.h"
+
+using namespace ParabolicRamp;
+
+namespace aikido {
+namespace planner {
+namespace parabolic {
+namespace hauserParabolicSmoother {
+
+static constexpr double CHECK_RESOLUTION = 1e-3;
+
+class RampRandomGenerator : public RandomNumberGeneratorBase
+{
+public:
+    RampRandomGenerator(aikido::util::RNG::result_type _rngSeed)
+        : rng_(_rngSeed), distribution_(0.0, 1.0)
+    {
+    }
+
+    virtual Real Rand() { return distribution_(rng_); }
+    aikido::util::RNGWrapper<std::mt19937> rng_;
+    std::uniform_real_distribution<>       distribution_;
+};
+
+class SmootherFeasibilityCheckerBase : public ParabolicRamp::FeasibilityCheckerBase
+{
+public:
+    SmootherFeasibilityCheckerBase(aikido::constraint::TestablePtr testable,
+                                   double checkResolution = CHECK_RESOLUTION)
+        : testable_(std::move(testable)),
+          checkResolution_(checkResolution)
+    {
+      interpolator_ = std::make_shared<aikido::statespace::GeodesicInterpolator>
+              (testable_->getStateSpace());
+    }
+
+    virtual bool ConfigFeasible(ParabolicRamp::Vector const &x)
+    {
+      statespace::StateSpacePtr stateSpace = testable_->getStateSpace();
+      Eigen::VectorXd eigX = toEigen(x);
+      auto state = stateSpace->createState();
+      stateSpace->expMap(eigX, state);
+      return testable_->isSatisfied(state);
+    }
+
+    virtual bool SegmentFeasible(ParabolicRamp::Vector const &a,
+                                 ParabolicRamp::Vector const &b)
+    {
+       statespace::StateSpacePtr stateSpace = testable_->getStateSpace();
+       Eigen::VectorXd eigA = toEigen(a);
+       Eigen::VectorXd eigB = toEigen(b);
+
+       auto testState = stateSpace->createState();
+       auto startState = stateSpace->createState();
+       auto goalState = stateSpace->createState();
+       stateSpace->expMap(eigA, startState);
+       stateSpace->expMap(eigB, goalState);
+
+       aikido::util::VanDerCorput vdc{1, true, true, checkResolution_};
+
+       for (const auto alpha : vdc)
+       {
+         interpolator_->interpolate(startState, goalState, alpha, testState);
+         if (!testable_->isSatisfied(testState))
+         {
+           return false;
+         }
+       }
+       return true;
+    }
+
+private:
+    aikido::constraint::TestablePtr                           testable_;
+    double                                                    checkResolution_;
+    std::shared_ptr<aikido::statespace::GeodesicInterpolator> interpolator_;
+};
+
+bool needsBlend(ParabolicRamp::ParabolicRampND const &rampNd)
+{
+    for (size_t idof = 0; idof < rampNd.dx1.size(); ++idof)
+    {
+        if (std::fabs(rampNd.dx1[idof]) > ParabolicRamp::EpsilonV)
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool tryBlend(ParabolicRamp::DynamicPath& dynamicPath,
+              ParabolicRamp::RampFeasibilityChecker feasibilityChecker,
```

Comment:
Only changed it to ParabolicRamp::RampFeasibilityChecker&. 
Because it calls ```dynamicPath.TryShortcut(t1, t2, feasibilityChecker);``` , which is ``` bool TryShortcut(Real t1,Real t2,RampFeasibilityChecker& check); ```